bootstrap/kubeadm/internal/bottlerocket/bottlerocket.go

```diff
@@ -211,6 +215,11 @@ func getBottlerocketNodeUserData(bootstrapContainerUserData []byte, users []boot
 		bottlerocketInput.RegistryMirrorCACert = base64.StdEncoding.EncodeToString([]byte(config.RegistryMirrorConfiguration.CACert))
 	}
 
+	if config.RegistryMirrorCredentials != nil {
```

Comment:
We might have to change this to check against an empty string for username/password